spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binding/StreamListenerMethodUtils.java

```diff
@@ -61,7 +61,7 @@ protected static int outputAnnotationCount(Method method) {
 
 	protected static void validateStreamListenerMethod(Method method, int inputAnnotationCount,
 			int outputAnnotationCount, String methodAnnotatedInboundName, String methodAnnotatedOutboundName,
-			boolean isDeclarative) {
+			boolean isDeclarative, StreamListener streamListener) {
```

Comment:
Given most of the other attributes are also derived from `StreamListener` annotation, may be we can just pass the `condition` String value here instead of the annotation itself?